### PR TITLE
Ddcnls 5981

### PR DESCRIPTION
**Pull request by brendanstraughan**

### CODE REVIEW

File: app/views/ambiguousjourney/deadlineIs.scala.html [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 @main(title = Messages("label.you_need_to_register_to_do__your_tax_return_online"),
   pageName = Some(Messages("label.your_personal_tax_account")),
-  backlinkUrl = Some(controllers.routes.AmbiguousJourneyController.usedUtrToRegisterChoice().url)
```
Comment (kurtlogan) [inline app/views/ambiguousjourney/deadlineIs.scala.html]: This shouldn't have changed

File: app/controllers/AmbiguousJourneyController.scala [hunk 2]
```diff
@@ -221,7 +221,7 @@ class AmbiguousJourneyController @Inject() (
           case "deadline" => Ok(views.html.ambiguousjourney.deadlineIs(saUtr, continueUrl))
           case "letter-in-post" => Ok(views.html.ambiguousjourney.letterMayBeInPost(saUtr, continueUrl))
           case "wrong-account" => Ok(views.html.ambiguousjourney.wrongAccount(saUtr, continueUrl, routes.AmbiguousJourneyController.usedUtrToEnrolChoice()))
-          case "pin-expired-register" => Ok(views.html.ambiguousjourney.wrongAccount(saUtr, continueUrl, routes.AmbiguousJourneyController.usedUtrToRegisterChoice()))
```
Comment (kurtlogan) [inline app/controllers/AmbiguousJourneyController.scala]: This shouldn't have changed

File: app/controllers/AmbiguousJourneyController.scala [hunk 3]
```diff
@@ -112,35 +112,35 @@ class AmbiguousJourneyController @Inject() (
...
-              else
-                Future.successful(Redirect(routes.AmbiguousJourneyController.receivedUtrLetterChoice()))
-            }
+    AmbiguousUserFlowDto.form.bindFromRequest.fold(
+      formWithErrors => {
+        Future.successful(BadRequest(views.html.ambiguousjourney.filedReturnByPostChoice(formWithErrors)))
+      },
+      ambiguousFiledOnlineChoiceDto => {
+        ambiguousFiledOnlineChoiceDto.value match {
+          case true => Future.successful(Redirect(routes.AmbiguousJourneyController.usedUtrToEnrolChoice()))
```
Comment (kurtlogan) [inline app/controllers/AmbiguousJourneyController.scala]: This route shouldn't have changed

File: conf/prod.routes [hunk 4]
```diff
@@ -29,8 +29,8 @@ GET    /personal-account/self-assessment/received-utr-letter                 @co
 POST   /personal-account/self-assessment/received-utr-letter                 @controllers.AmbiguousJourneyController.processReceivedUtrLetterChoice
 GET    /personal-account/self-assessment/have-you-enrolled                   @controllers.AmbiguousJourneyController.usedUtrToEnrolChoice
 POST   /personal-account/self-assessment/have-you-enrolled                   @controllers.AmbiguousJourneyController.processUsedUtrToEnrolChoice
-GET    /personal-account/self-assessment/used-utr-to-register                @controllers.AmbiguousJourneyController.usedUtrToRegisterChoice
```
Comment (kurtlogan) [inline conf/prod.routes]: This shouldn't have changed

File: conf/prod.routes [hunk 5]
```diff
@@ -29,8 +29,8 @@ GET    /personal-account/self-assessment/received-utr-letter                 @co
...
-POST   /personal-account/self-assessment/used-utr-to-register                @controllers.AmbiguousJourneyController.processUsedUtrToRegisterChoice
```
Comment (kurtlogan) [inline conf/prod.routes]: This shouldn't have changed

File: test/controllers/AmbiguousJourneyControllerSpec.scala [hunk 6]
```diff
@@ -293,36 +293,36 @@ class AmbiguousJourneyControllerSpec extends BaseSpec with ViewSpec {
     }
   }
 
-  "Calling AmbiguousJourneyController.processUsedUtrToRegisterChoice" should {
```
Comment (kurtlogan) [inline test/controllers/AmbiguousJourneyControllerSpec.scala]: This shouldn't have changed

File: test/controllers/AmbiguousJourneyControllerSpec.scala [hunk 7]
```diff
@@ -293,36 +293,36 @@ class AmbiguousJourneyControllerSpec extends BaseSpec with ViewSpec {
...
+  "Calling AmbiguousJourneyController.processUsedUtrToEnrolChoice" should {
 
     "redirect to 'Your pin has expired' page when supplied with value Yes (true)" in new LocalSetupJourney {
-      val r = controller.processUsedUtrToRegisterChoice(buildFakeRequestWithAuth("POST").withFormUrlEncodedBody("ambiguousUserFormChoice" -> "true"))
```
Comment (kurtlogan) [inline test/controllers/AmbiguousJourneyControllerSpec.scala]: This shouldn't have changed

File: test/controllers/AmbiguousJourneyControllerSpec.scala [hunk 8]
```diff
@@ -293,36 +293,36 @@ class AmbiguousJourneyControllerSpec extends BaseSpec with ViewSpec {
...
+      val r = controller.processUsedUtrToEnrolChoice(buildFakeRequestWithAuth("POST").withFormUrlEncodedBody("ambiguousUserFormChoice" -> "true"))
       override lazy val getSelfAssessmentServiceResponse = AmbiguousFilerSelfAssessmentUser(SaUtr("1111111111"))
 
       status(r) shouldBe SEE_OTHER
       redirectLocation(await(r)) shouldBe Some("/personal-account/self-assessment/result/pin-expired-register")
     }
 
     "redirect to 'The deadline is' page when supplied with value No (false)" in new LocalSetupJourney {
-      val r = controller.processUsedUtrToRegisterChoice(buildFakeRequestWithAuth("POST").withFormUrlEncodedBody("ambiguousUserFormChoice" -> "false"))
```
Comment (kurtlogan) [inline test/controllers/AmbiguousJourneyControllerSpec.scala]: This shouldn't have changed

File: test/controllers/AmbiguousJourneyControllerSpec.scala [hunk 9]
```diff
@@ -293,36 +293,36 @@ class AmbiguousJourneyControllerSpec extends BaseSpec with ViewSpec {
...
+      val r = controller.processUsedUtrToEnrolChoice(buildFakeRequestWithAuth("POST").withFormUrlEncodedBody("ambiguousUserFormChoice" -> "false"))
       override lazy val getSelfAssessmentServiceResponse = AmbiguousFilerSelfAssessmentUser(SaUtr("1111111111"))
 
       status(r) shouldBe SEE_OTHER
       redirectLocation(await(r)) shouldBe Some("/personal-account/self-assessment/result/deadline")
     }
 
     "return a bad request when supplied no value" in new LocalSetupJourney {
-      val r = controller.processUsedUtrToRegisterChoice(buildFakeRequestWithAuth("POST"))
```
Comment (kurtlogan) [inline test/controllers/AmbiguousJourneyControllerSpec.scala]: This shouldn't have changed

File: test/controllers/AmbiguousJourneyControllerSpec.scala [hunk 10]
```diff
@@ -293,36 +293,36 @@ class AmbiguousJourneyControllerSpec extends BaseSpec with ViewSpec {
...
       override lazy val getSelfAssessmentServiceResponse = AmbiguousFilerSelfAssessmentUser(SaUtr("1111111111"))
 
       status(r) shouldBe BAD_REQUEST
     }
 
   }
 
-  "Calling AmbiguousJourneyController.usedUtrToRegisterChoice" should {
-    "return 200 when AmbiguousJourneyController.usedUtrToRegisterChoice is called" in new LocalSetupJourney {
-      val r = controller.usedUtrToRegisterChoice(buildFakeRequestWithAuth("GET"))
```
Comment (kurtlogan) [inline test/controllers/AmbiguousJourneyControllerSpec.scala]: This shouldn't have changed

File: conf/prod.routes (hunk 4 above)
Comment (brendanstraughan) [reply conf/prod.routes]: done

File: app/views/ambiguousjourney/deadlineIs.scala.html (hunk 1 above)
Comment (brendanstraughan) [reply app/views/ambiguousjourney/deadlineIs.scala.html]: done

File: app/controllers/AmbiguousJourneyController.scala (hunk 3 above)
Comment (brendanstraughan) [reply app/controllers/AmbiguousJourneyController.scala]: done

File: conf/prod.routes (hunk 5 above)
Comment (brendanstraughan) [reply conf/prod.routes]: done

File: test/controllers/AmbiguousJourneyControllerSpec.scala (hunk 6 above)
Comment (brendanstraughan) [reply test/controllers/AmbiguousJourneyControllerSpec.scala]: done

File: test/controllers/AmbiguousJourneyControllerSpec.scala (hunk 7 above)
Comment (brendanstraughan) [reply test/controllers/AmbiguousJourneyControllerSpec.scala]: done

File: test/controllers/AmbiguousJourneyControllerSpec.scala (hunk 8 above)
Comment (brendanstraughan) [reply test/controllers/AmbiguousJourneyControllerSpec.scala]: done

File: test/controllers/AmbiguousJourneyControllerSpec.scala (hunk 9 above)
Comment (brendanstraughan) [reply test/controllers/AmbiguousJourneyControllerSpec.scala]: done

File: test/controllers/AmbiguousJourneyControllerSpec.scala (hunk 10 above)
Comment (brendanstraughan) [reply test/controllers/AmbiguousJourneyControllerSpec.scala]: done

File: app/controllers/AmbiguousJourneyController.scala [hunk 11]
```diff
@@ -112,35 +112,35 @@ class AmbiguousJourneyController @Inject() (
...
-        ambiguousFiledOnlineChoiceDto => {
-          ambiguousFiledOnlineChoiceDto.value match {
-            case true => Future.successful(Redirect(routes.AmbiguousJourneyController.usedUtrToRegisterChoice()))
-            case false => {
-              if (configDecorator.saAmbigSkipUTRLetterEnabled)
-                Future.successful(Redirect(routes.AmbiguousJourneyController.usedUtrToEnrolChoice()))
-              else
-                Future.successful(Redirect(routes.AmbiguousJourneyController.receivedUtrLetterChoice()))
-            }
+    AmbiguousUserFlowDto.form.bindFromRequest.fold(
```
Comment (kurtlogan) [inline app/controllers/AmbiguousJourneyController.scala]: Only change here is the tabbing, can you undo this change

File: app/controllers/AmbiguousJourneyController.scala [hunk 12]
```diff
@@ -112,35 +112,35 @@ class AmbiguousJourneyController @Inject() (
...
+      }
+    )
   }
-
-  def usedUtrToRegisterChoice: Action[AnyContent] = VerifiedAction(baseBreadcrumb) { implicit pertaxContext =>
-    enforceAmbiguousUser {_ =>
-      Future.successful(Ok(views.html.ambiguousjourney.usedUtrToRegisterChoice(AmbiguousUserFlowDto.form)))
+    def usedUtrToRegisterChoice: Action[AnyContent] = VerifiedAction(baseBreadcrumb) { implicit pertaxContext =>
+      enforceAmbiguousUser {_ =>
+        Future.successful(Ok(views.html.ambiguousjourney.usedUtrToEnrolChoice(AmbiguousUserFlowDto.form, controllers.routes.AmbiguousJourneyController.filedReturnByPostChoice().url)))
```
Comment (kurtlogan) [inline app/controllers/AmbiguousJourneyController.scala]:
```suggestion
        Future.successful(Ok(views.html.ambiguousjourney.usedUtrToEnrolChoice(AmbiguousUserFlowDto.form, routes.AmbiguousJourneyController.filedReturnByPostChoice().url)))
```

File: app/controllers/AmbiguousJourneyController.scala [hunk 13]
```diff
@@ -112,35 +112,35 @@ class AmbiguousJourneyController @Inject() (
...
     }
-  }
+
 
   def processUsedUtrToRegisterChoice: Action[AnyContent] = VerifiedAction(baseBreadcrumb) {
     implicit pertaxContext =>
       AmbiguousUserFlowDto.form.bindFromRequest.fold(
         formWithErrors => {
-          Future.successful(BadRequest(views.html.ambiguousjourney.usedUtrToRegisterChoice(formWithErrors)))
+          Future.successful(BadRequest(views.html.ambiguousjourney.usedUtrToEnrolChoice(formWithErrors, controllers.routes.AmbiguousJourneyController.filedReturnByPostChoice().url)))
```
Comment (kurtlogan) [inline app/controllers/AmbiguousJourneyController.scala]:
```suggestion
          Future.successful(BadRequest(views.html.ambiguousjourney.usedUtrToEnrolChoice(formWithErrors, routes.AmbiguousJourneyController.filedReturnByPostChoice().url)))
```

File: app/controllers/AmbiguousJourneyController.scala [hunk 14]
```diff
@@ -221,7 +221,7 @@ class AmbiguousJourneyController @Inject() (
           case "deadline" => Ok(views.html.ambiguousjourney.deadlineIs(saUtr, continueUrl))
           case "letter-in-post" => Ok(views.html.ambiguousjourney.letterMayBeInPost(saUtr, continueUrl))
           case "wrong-account" => Ok(views.html.ambiguousjourney.wrongAccount(saUtr, continueUrl, routes.AmbiguousJourneyController.usedUtrToEnrolChoice()))
-          case "pin-expired-register" => Ok(views.html.ambiguousjourney.wrongAccount(saUtr, continueUrl, routes.AmbiguousJourneyController.usedUtrToRegisterChoice()))
+          case "pin-expired-register" => Ok(views.html.ambiguousjourney.wrongAccount(saUtr, continueUrl, routes.AmbiguousJourneyController.usedUtrToRegisterChoice))
```
Comment (kurtlogan) [inline app/controllers/AmbiguousJourneyController.scala]: Only change here is the parenthesis, undo change as it doesn't add anything to this pr

File: app/controllers/AmbiguousJourneyController.scala (hunk 11 above)
Comment (brendanstraughan) [reply app/controllers/AmbiguousJourneyController.scala]: done

File: app/controllers/AmbiguousJourneyController.scala (hunk 13 above)
Comment (brendanstraughan) [reply app/controllers/AmbiguousJourneyController.scala]: done

File: app/controllers/AmbiguousJourneyController.scala (hunk 12 above)
Comment (brendanstraughan) [reply app/controllers/AmbiguousJourneyController.scala]: done

File: app/controllers/AmbiguousJourneyController.scala (hunk 14 above)
Comment (brendanstraughan) [reply app/controllers/AmbiguousJourneyController.scala]: done
